src/compiler/checker.ts

```diff
@@ -7227,12 +7227,31 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 anyType : getNonMissingTypeOfSymbol(propertySymbol);
             const saveEnclosingDeclaration = context.enclosingDeclaration;
             context.enclosingDeclaration = undefined;
-            if (context.tracker.canTrackSymbol && isLateBoundName(propertySymbol.escapedName)) {
-                if (propertySymbol.declarations) {
-                    const decl = first(propertySymbol.declarations);
+            const decl = propertySymbol.declarations && first(propertySymbol.declarations);
+

```

Comment:
I wanted to be able to force the compiler to preserve the computed property expressions in types instead of resolving them, if the computed property is an accessible entity name (identifier, or identifier chain):

So for example for: 

```ts
const A = 1;
const E = { A: 2 } as const
let o = { [A]: 1, [E.A]: 2 }
```
The checker would always produce as the type of `o`: `{ [1]: number, [2]: number }` which is correct but not a type a dev would ever write. With this change, if `A` and `E.A` are accessible in the current scope (either are already in the scope or can be imported), they will be preserved and we get `{ [A]: number, [E.A]: number }`